src/ru/nsu/ccfit/zuev/osu/game/GameObject.java

```diff
@@ -51,7 +51,9 @@ public boolean isStartHit(){
         return startHit;
     }
 
-    public void tryHit(float dt) {return;};
+    public boolean tryHit(float dt) {return false;};
```

Comment:
I know this is an old code, but this method is overridden in any class that inherits `GameObject`. Maybe make it `abstract` instead?

```suggestion
    public abstract boolean tryHit(float dt);
```

Alternatively, you can add a comment explaining why it returns `false` because I'm confused.

Reply:
This method is only used in HitCircle/Slider/Spinner. In other GameObject, this method is useless. so I don't use abstract.
Maybe It is better to refactor inheritance relationships, but I don't have time
I'll add a comment in a moment